### PR TITLE
feat: jwt profile grant

### PR DESCRIPTION
**Pull request by livio-a**

### CODE REVIEW

File: pkg/rp/tockenexchange.go [hunk 1]
```diff
@@ -25,3 +31,66 @@ type DelegationTokenExchangeRP interface {
...
+	jsonadsk, err := json.Marshal(assertion)
+	if err != nil {
+		return "", err
+	}
+	signiert, err := signer.Sign(jsonadsk)
+	if err != nil {
+		return "", err
+	}
+	return signiert.CompactSerialize()
+}
```
Comment (adlerhurst) [inline pkg/rp/tockenexchange.go:78]:
```suggestion
	marshalledAssertion, err := json.Marshal(assertion)
	if err != nil {
		return "", err
	}
	signedAssertion, err := signer.Sign(marshalledAssertion)
	if err != nil {
		return "", err
	}
	return signedAssertion.CompactSerialize()
}
```

**Review by adlerhurst — APPROVED** [review]

**Review by adlerhurst — APPROVED** [review]

**Review by adlerhurst — APPROVED** [review]

File: pkg/oidc/verifier.go [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+	Offset() time.Duration
+}
+
+type verifierConfig struct {
+	issuer           string
+	clientID         string
+	nonce            string
+	ignoreAudience   bool
+	ignoreExpiration bool
+	//iat               *iatConfig
```
Comment (fgerschwiler) [inline pkg/oidc/verifier.go:62]: remove unused code

File: pkg/oidc/verifier.go [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+	acr               ACRVerifier
+	maxAge            time.Duration
+	supportedSignAlgs []string
+
+	// httpClient *http.Client
```
Comment (fgerschwiler) [inline pkg/oidc/verifier.go:67]: remove unused code

File: pkg/op/op.go [hunk 4]
```diff
@@ -41,18 +72,318 @@ func CreateRouter(o OpenIDProvider, interceptors ...HttpInterceptor) *mux.Router
...
+	CryptoKey                [32]byte
+	DefaultLogoutRedirectURI string
+	CodeMethodS256           bool
+	// ScopesSupported:                   oidc.SupportedScopes,
+	// ResponseTypesSupported:            responseTypes,
+	// GrantTypesSupported:               oidc.SupportedGrantTypes,
+	// ClaimsSupported:                   oidc.SupportedClaims,
+	// IdTokenSigningAlgValuesSupported:  []string{keys.SigningAlgorithm},
+	// SubjectTypesSupported:             []string{"public"},
+	// TokenEndpointAuthMethodsSupported:
```
Comment (fgerschwiler) [inline pkg/op/op.go:98]: remove if not used

Comment (livio-a) [reply pkg/op/op.go:100]: added ```TODO: add to config after updating Configuration interface for DiscoveryConfig```

**Review by fgerschwiler — APPROVED** [review]

**Pull request merged**